### PR TITLE
integration with Yealink headsets, use webhid

### PR DESCRIPTION
**Pull request by Jaycan-YL**

### CODE REVIEW

**Comment by djhouseknecht** [comment]
I'm going to defer the finally approval to @jensengar and @maxwellmooney13 since they did most of the work on this library. 

But could you add unit tests, @Jaycan-YL? We like to keep all our open source libraries up to 100% coverage where possible.

File: react-app/src/library/services/vendor-implementations/yealink/yealink.ts [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+      this.isHold = false;
+    }
+
+    if (value & this.muteFlag) {
+      this.isMuted = true;
+    } else {
+      this.isMuted = false;
+    }
+
+    this.logger.info(`send to dev ${value}`);
```
Comment (jensengar) [inline react-app/src/library/services/vendor-implementations/yealink/yealink.ts:299]: this should be a `debug` level log.

**Review by jensengar — CHANGES_REQUESTED** [review]
Nothing major, mostly just a few styling tweaks. We are in the process of ordering some headsets so we can do a sanity test on it, then we'll get it in. Thanks contributing!

**Comment by maxwellmooney13** [comment]
I have tested with five different YeaLink devices (UH38 Dual UC, UH36 Mono, UH34 Dual, WH62 Mono UC, WH66 Dual UC) and everything seems to work great!  I did notice with the WH62 that when I ended a call at one point, it also muted the call.  Not a huge deal but it is something I saw and was definitely strange to see.  Once Garrett's comments are addressed, you guys should be good to go! Thank you for your time, effort and patience in this.

Sidenote: We recently had an issue with a feature we offer for "auto answering calls".  I included functionality for this in a recent PR and it might be a good thing to quickly test and ensure auto answer works as expected for your devices as well.

**Comment by maxwellmooney13** [comment]
Did a very quick check with the auto answer capabilities and it seems fine from that perspective!  Might be good to do a quick double check on that though.  Otherwise, looks good!

**Review by jensengar — APPROVED** [review]

**Pull request merged**